cmd/kube-proxy/app/server.go

```diff
@@ -465,6 +467,13 @@ addon that provides cluster DNS for these cluster IPs. The user must create a se
 with the apiserver API to configure the proxy.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			verflag.PrintAndExitIfRequested()
+
+			// Config and flags parsed, now we can initialize logging.
```

Comment:
this comment seems to have been copied from the kubelet sources but it doesn't actually seem to apply here?

I'm not sure this is the right place for the InitLogs, etc, code either. How did you decide to put it here as opposed to anywhere else?

Reply:
Yes, copied here from the kubelet sources. `InitLogs` should be called after parsing flags and before cliflag.PrintFlags. 

Refer to:https://github.com/kubernetes/kubernetes/blob/b32941f60d3ba789965bbf32f4f5bf756feab075/staging/src/k8s.io/component-base/logs/logs.go#L149
https://github.com/kubernetes/kubernetes/blob/b32941f60d3ba789965bbf32f4f5bf756feab075/cmd/kube-proxy/app/server.go#L468

I went through the startup source code of kube-apiserver, kube-controller-manager, kube-scheduler, and kubelet.

Let's take kube-apiserver as an example:

- New log options: https://github.com/kubernetes/kubernetes/blob/b32941f60d3ba789965bbf32f4f5bf756feab075/cmd/kube-apiserver/app/options/options.go#L107

- Add log flag: https://github.com/kubernetes/kubernetes/blob/b32941f60d3ba789965bbf32f4f5bf756feab075/cmd/kube-apiserver/app/options/options.go#L156

- ValidateAndApply log options: https://github.com/kubernetes/kubernetes/blob/b32941f60d3ba789965bbf32f4f5bf756feab075/cmd/kube-apiserver/app/server.go#L119

kube-proxy is similar to kubelet but different from the other components. so here is the same as kubelet. Please see https://github.com/kubernetes/kubernetes/pull/105797/files#diff-3a14a9838132c6668f4d100c26afa05b471b149c469edfece9b84fcd058c8a4b


Maybe this needs to be ack by sig/instrumentation reviewer

/cc @pohly @liggitt

Reply:
The config file is not parsed until a few lines later (from `opts.Complete`)